test/parallel/test-console-table.js

```diff
@@ -0,0 +1,77 @@
+'use strict';
+
+require('../common');
+
+const assert = require('assert');
+const { Console } = require('console');
+
+const queue = [];
+
+const console = new Console({ write: (x) => {
+  queue.push(x);
+}, removeListener: () => {} }, process.stderr, false);
+
+function test(x, y, z) {
+  let args = [x];
+  let expected = y;
+  if (arguments.length === 3) {
+    args = [x, y];
+    expected = z;
+  }
+  console.table(...args);
+  assert.strictEqual(queue.shift(), expected);
+}
+
+/* eslint-disable no-tabs */
+
+test({ a: 1 }, `(index)	Values
+a	1
+`);
+
+test([{ a: 1 }, { b: 2 }], `(index)	a	b
+0	1	undefined
+1	undefined	2
+`);
+
+{
+  const map = new Map([ [1, 2], [3, 4] ]);
+  const map2 = new Map([ [1, 3], [2, 4] ]);
+  test([map, map2], `(index)	1	3	2
+0	2	4	undefined
+1	3	undefined	4
+`);
+}
+
+test([3, 4, { a: 3 }], `(index)	a
+0	undefined
+1	undefined
+2	3
+`);
+
+test([3, 4, { a: 3 }], ['nonexistent'], `(index)	a
```

Comment:
Chrome shows this as the following that I think would be great to have:
 
```
(index) Value a
0 3
1 4
2   3
```

Reply:
i don't understand the output of that table

Reply:
There is no `undefined` value for `a` in index 0 and 1. The value does not exist. But there is a value and we can show that with `3` and `4`.

Just enter `console.table([3, 4, { a: 3 }])` in chrome to see the output :-)

Reply:
rn the only disparity seems to be that i choose to ignore filters of props that don't exist while the browsers use `undefined`. i can change that.
![](https://gc.gy/濫.png)

Reply:
This is how it looks like for me:

![image](https://user-images.githubusercontent.com/8822573/36336134-74b109d0-1385-11e8-9410-3023a5f59579.png)
![image](https://user-images.githubusercontent.com/8822573/36336138-8245946c-1385-11e8-912f-280ce89af14d.png)

Reply:
@BridgeAR There isn’t really a standard on what should be printed. As long as the output makes sense, which IMO it does, it’s good.

Reply:
@TimothyGu I am aware that there is no standard for it but I feel like the output used in chrome does make more sense and provides a relatively good user experience.

Reply:
I'm in favour of the change requested by @BridgeAR here, in relation to missing vs `undefined` values. I think it would go a long way towards making this feature more useful.